backend/src/LibExecutionStdLib/StdLib.fs

```diff
@@ -76,11 +76,16 @@ let infixFnMapping : Map<FQFnName.StdlibFnName, (FQFnName.StdlibFnName * Depreca
     ("String", "append", 1), (("", "++"), NotDeprecated)
     ("", "equals", 0), (("", "=="), NotDeprecated)
     ("", "notEquals", 0), (("", "!="), NotDeprecated)
-    // TODO: deprecate when there's a nice way to move from infix fn to language version
-    // ("Bool", "and", 0), (("", "&&"), DeprecatedBecause("Use `&&` instead"))
-    // ("Bool", "or", 0), (("", "||"), DeprecatedBecause("Use `||` instead"))
-    ("Bool", "and", 0), (("", "&&"), NotDeprecated)
-    ("Bool", "or", 0), (("", "||"), NotDeprecated) ]
+    ("Bool", "and", 0),
+    (("", "&&"),
+     DeprecatedBecause(
+       "Use built-in keyword `&&` instead. Use the `convert-to-short-circuiting` command to convert automatically"
```

Comment:
I think we should call out what "short-circuiting" is _here_ (as well as in the command) - I'm not sure if "short-circuiting" is a known term to the majority of programmers.

Reply:
I do think we should do better to explain this concept and others, but I'd like to wait for the doc box project to finish before we iterate on this, so we can think a little bigger.